packages/colorpickers/src/elements/ColorPicker/index.tsx

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, {
+  useEffect,
+  useCallback,
+  useContext,
+  useReducer,
+  forwardRef,
+  HTMLAttributes,
+  useMemo,
+  useRef
+} from 'react';
+import PropTypes from 'prop-types';
+import { ThemeContext } from 'styled-components';
+import { Label } from '@zendeskgarden/react-forms';
+import { ColorWell } from './ColorWell';
+import {
+  StyledHue,
+  StyledSliderGroup,
+  StyledLabel,
+  StyledInput,
+  StyledAlpha,
+  StyledSliders,
+  StyledHexField,
+  StyledRGBAField,
+  StyledInputGroup,
+  StyledPreview,
+  StyledColorPicker,
+  StyledCheckered,
+  StyledAlphaGradient,
+  StyledAlphaField,
+  StyledHueField
+} from '../../styled';
+import { areColorsEqual, convertStringToColor, getInitialState, reducer } from './reducer';
+import { IColor, IHSVColor } from '../../utils/types';
+
+export interface IColorPickerLabels {
+  hueSlider?: string;
+  alphaSlider?: string;
+  hex?: string;
+  red?: string;
+  green?: string;
+  blue?: string;
+  alpha?: string;
+}
+
+export interface IColorPickerProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, 'color' | 'onChange'> {
+  /** A default color for a uncontrolled color picker. Can be a valid color string or IColor state */
+  defaultColor?: string | IColor;
+  /** A color value for a controlled color picker. Can be a valid color string or IColor state */
+  color?: string | IColor;
+  /**
+   * Handles color picker changes
+   *
+   * @param {Object} color A color picker state
+   */
+  onChange?: (color: IColor) => void;
+  /** Replaces the default labels within the color picker */
```

Comment:
*Replaces label text within the color picker

Is there an effective way to help docgen pick up the interface structure here? Showing an interface that the consumer doesn't have access to in the docs isn't super helpful.

Reply:
Updated the description copy. Inlining the `labels` type results in a detailed type:

<img width="772" alt="Screen Shot 2021-02-24 at 9 17 04 AM" src="https://user-images.githubusercontent.com/1811365/109039350-8ccc7380-7681-11eb-971a-49f104956b5e.png">

Alternatively, there could be a link in the description that links directly to the line of code where `IColorPickerLabels` is defined.

Reply:
I think a link is likely to fall out-of-date over time. Better to provide the information inline.